Src/EventCounterCollector/EventCounterCollector/EventCounterListener.cs

```diff
@@ -0,0 +1,232 @@
+                    }
+                }
+                else
+                {
+                    EventCounterCollectorEventSource.Log.IgnoreEventWrittenAsEventSourceNotInConfiguredList(eventData.EventSource.Name);
+                }
+            }
+            else
+            {
+                EventCounterCollectorEventSource.Log.IgnoreEventWrittenAsNotInitialized(eventData.EventSource.Name);
+            }
+        }
+
+        private void EnableIfRequired(EventSource eventSource)
+        {
+            try
+            {
+                // The EventSourceName is in the list we want to collect some counters from.
+                if (this.countersToCollect.ContainsKey(eventSource.Name))
+                {
+                    Dictionary<string, string> refreshInterval = new Dictionary<string, string>();
+                    // 60 sec is hardcoded and not allowed for user customization as backend expects 1 min aggregation.
+                    // TODO: Need to revisit if this should be changed.               
+                    refreshInterval.Add("EventCounterIntervalSec", this.refreshIntervalInSecs);
+
+                    // Unlike regular Events, the only relevant parameter here for EventCounter is the dictionary containing EventCounterIntervalSec.
+                    this.EnableEvents(eventSource, this.level, (EventKeywords)(-1), refreshInterval);
+
+                    EventCounterCollectorEventSource.Log.EnabledEventSource(eventSource.Name);
+                }
+                else
+                {
+                    EventCounterCollectorEventSource.Log.NotEnabledEventSource(eventSource.Name);
+                }
+            }
+            catch (Exception ex)
+            {
+                EventCounterCollectorEventSource.Log.EventCounterCollectorError("EventCounterListener EnableEventSource", ex.Message);
+            }
+        }
+
+        private void ExtractAndPostMetric(string eventSourceName, IDictionary<string, object> eventPayload)
+        {
+            try
+            {
+                MetricTelemetry metricTelemetry = new MetricTelemetry();
+                bool calculateRate = false;
+                double actualValue = 0.0;
+                double actualInterval = 0.0;
+                int actualCount = 0;
+                foreach (KeyValuePair<string, object> payload in eventPayload)
+                {
+                    var key = payload.Key;
+                    if (key.Equals("Name", StringComparison.OrdinalIgnoreCase))
+                    {
+                        var counterName = payload.Value.ToString();
+                        if (this.countersToCollect[eventSourceName].Contains(counterName))
+                        {
+                            metricTelemetry.Name = string.Format(CultureInfo.InvariantCulture, "{0}|{1}", eventSourceName, counterName);
```

Comment:
it is cheaper to concatenate strings than call string.format. We are on the hot path here

http://geekswithblogs.net/BlackRabbitCoder/archive/2010/05/10/c-string-compares-and-concatenations.aspx

Reply:
Not sure what is the recommendation here..https://stackoverflow.com/questions/296978/when-is-it-better-to-use-string-format-vs-string-concatenation
I am unable to find concrete recommendation about one vs other.